flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/table/TableSinkITCase.scala

```diff
@@ -635,6 +635,58 @@ class TableSinkITCase extends StreamingTestBase {
     assertEquals(expected.sorted, result.sorted)
   }
 
+  @Test
```

Comment:
I think we could write these tests using `TableFactoryHarness`. This would avoid Scala and using these sources that require static state. Just a thought, though, and maybe something to consider for how to utilize this infrastructure going forward.

Reply:
Yep, makes sense, I will wait for https://github.com/apache/flink/pull/17759 and implement the test in `CommonExecSinkITCase`.